### ISSUE TITLE
[SPA/StyleSystem] Support for inclusion of Stylesystem props in Sling Models

### ISSUE THREAD

**Issue by friendlymahi** [adobe/aem-core-wcm-components#1374]
## Feature Request

**Is your feature request related to a problem? Please describe.**
As of date, leveraging SPAEditor integration comes with limitations like not being able to use Style System capability of AEM, which is very powerful for managing the styles library one may load into the policies. And Core Components on the other hand are a great feature no one wants to rebuild as new components. So to start with we need ability for Core Components to be able to export additional styles to apply, via Sling Models, so it is up to the consumer to leverage that integration. And its for this reason we see a need for this enhancement in this repo.

**Describe the solution you'd like**
Component interface that we have today in Core Components provides methods like getData to support data layer. Similarly we can provide getStyles or getStylesToAttach for each component, and include the logic to bind component policy and read the styles applied using dialog. 

**Are there alternatives?**
We may be forced to re-do every sling model using the delegation pattern which is not ideal especially when we know we can resolve this issue from one place which shall eventually serve all. We have sample code that is used in this alternative that can go just in one place so there is no need for customizing sling models or write brand new unit tests when using the delegation pattern for things that are inevitable

**Documentation**
As with normal pages (non SPA), we use the paint brush to select the styles. And now its for the React/Angular components obtain this metadata for styles and wrap their components accordingly. Here is a sample snippet. We exposed sling model to return style props as **additionalStyles**, and in React we use **className** attribute to map the value to.
```
import React from 'react';
import { Heading } from 'my-ui-controls';
import './style.scss'

export default (props) => {
    const renamedProps = (({ type: tag = 'h1', additionalStyles: className = "headerdefault", ...remainingProps }) => ({ tag, className, ...remainingProps }))(props);
    return <div className="dbp-cmp-title" >
        <Heading {...renamedProps} />
    </div>
}
```

**Comment by friendlymahi** [comment]
@joerghoh @godanny86 -Any thoughts on this? I am tagging you as I saw your inputs very helpful in the past. Wondering if I can contribute to this. Thank you

**Comment by pfauchere** [comment]
Hi,

Thank you very much for the input.

We captured the need and intend to explore the topic soon.

That said, and if we are able to decorate Sling Models when they are being exported, I am not certain that the contribution will apply to the current repository. I cannot share or confirm any architecture decision as the work hasn't started yet.

Internal Ref: CQ-4292931

**Comment by friendlymahi** [comment]
@pfauchere - Thank you so much for the inputs. I already did send a pr and @vladbailescu is already helping with the review. However as I menyioned in my pr, idea behind is not to duplicate but to ensure all third party integrations can leverage the capability. Code change is very minor, and like you said change may have to go else where. Challenge I see is data-sly-resource or some other htl impl takes care of  assigning the styles in normal content pages, but the case will be different when using it in context of spa editor as this data needs to be available via Json. So unsure if the logic can be reused in both places for consuming the classes info, but will definitely come with some caveats. Thank you again and looking forward for inputs from you and @vladbailescu.
